safir/src/safir/metrics/_testing.py

```diff
@@ -0,0 +1,235 @@
+"""Helpers working with metrics events in unit tests."""
+
+from abc import ABC, abstractmethod
+from pprint import pformat
+from typing import Any, Generic, TypeVar
+from unittest.mock import ANY as MOCK_ANY
+
+from pydantic import BaseModel
+
+__all__ = [
+    "ANY",
+    "NOT_NONE",
+    "BaseAssertionError",
+    "NotPublishedConsecutivelyError",
+    "NotPublishedError",
+    "PublishedCountError",
+    "PublishedList",
+    "PublishedTooFewError",
+]
+
+P = TypeVar("P", bound=BaseModel)
```

Comment:
I don't think it matters a ton, but shouldn't the bound here be `EventPayload`, just to reduce the chance of errors from passing in some unrelated Pydantic model?